packages/vite/src/node/server/ws.ts

```diff
@@ -28,9 +28,10 @@ export function createWebSocketServer(
   let httpsServer: Server | undefined = undefined
 
   const hmr = isObject(config.server.hmr) && config.server.hmr
-  const wsServer =
-    (hmr && hmr.server) ||
-    ((!(hmr && hmr.port) || hmr.port !== config.server.port) && server)
+  const hmrServer = hmr && hmr.server
+  const hmrPort = hmr && hmr.port
+  const portsNotEqual = config.server.port && hmrPort !== config.server.port
+  const wsServer = hmrServer || ((!hmrPort || portsNotEqual) && server)
```

Comment:
Then this can become…
```suggestion
  const wsServer = hmrServer || (hmrPort !== config.server.port && server)
```
…and remove the `portsNotEqual` variable.